packages/mdc-tab/index.js

```diff
@@ -53,32 +74,62 @@ class MDCTab extends MDCComponent {
    * @return {!MDCTabFoundation}
    */
   getDefaultFoundation() {
-    return new MDCTabFoundation(/** @type {!MDCTabAdapter} */ (Object.assign({
-      setAttr: (attr, value) => this.root_.setAttribute(attr, value),
-      registerEventHandler: (evtType, handler) => this.root_.addEventListener(evtType, handler),
-      deregisterEventHandler: (evtType, handler) => this.root_.removeEventListener(evtType, handler),
-      addClass: (className) => this.root_.classList.add(className),
-      removeClass: (className) => this.root_.classList.remove(className),
-      hasClass: (className) => this.root_.classList.contains(className),
-    })));
+    return new MDCTabFoundation(
+      /** @type {!MDCTabAdapter} */ (Object.assign({
+        setAttr: (attr, value) => this.root_.setAttribute(attr, value),
+        registerEventHandler: (evtType, handler) => this.root_.addEventListener(evtType, handler),
+        deregisterEventHandler: (evtType, handler) => this.root_.removeEventListener(evtType, handler),
+        addClass: (className) => this.root_.classList.add(className),
+        removeClass: (className) => this.root_.classList.remove(className),
+        hasClass: (className) => this.root_.classList.contains(className),
+      },
+      this.getIndicatorAdapterMethods_())));
+  }
+
+  /**
+   * @return {!{
+   *   activateIndicator: function(!ClientRect): undefined,
+   *   deactivateIndicator: function(): undefined,
+   *   computeIndicatorClientRect: function(): !ClientRect,
+   * }}
+   */
+  getIndicatorAdapterMethods_() {
```

Comment:
I dont think you need to make this a separate method. Just move these three methods into the adapter class above.